readthedocs/config/config.py

```diff
@@ -745,12 +751,60 @@ def validate_build(self):
                 ),
             )
 
-        # Allow to override specific project
-        config_image = self.defaults.get('build_image')
-        if config_image:
-            build['image'] = config_image
+            # Allow to override specific project
+            config_image = self.defaults.get('build_image')
+            if config_image:
+                build['image'] = config_image
+
+        with self.catch_validation_error('build.apt_packages'):
+            raw_packages = self._raw_config.get('build', {}).get('apt_packages', [])
+            validate_list(raw_packages)
+            # Transform to a dict, so is easy to validate individual entries.
+            self._raw_config.setdefault('build', {})['apt_packages'] = (
+                list_to_dict(raw_packages)
+            )
+
+            build['apt_packages'] = [
+                self.validate_apt_package(index)
+                for index in range(len(raw_packages))
+            ]
+            if not raw_packages:
+                self.pop_config('build.apt_packages')
+
         return build
 
+    def validate_apt_package(self, index):
+        """
+        Validate the package name to avoid injections of extra options.
+
+        Packages names can be a regex pattern.
+        We just validate that they aren't interpreted as an option or file.
+        """
+        key = f'build.apt_packages.{index}'
+        package = self.pop_config(key)
+        with self.catch_validation_error(key):
+            validate_string(package)
+            package = package.strip()
+            invalid_starts = [
+                # Don't allow to inject extra options.
+                '-',
+                '\\',
+                # Don't allow to install from a path.
```

Comment:
Isn't this still possible by doing `repository/my/path/package.deb` as an item?

Reply:
https://www.debian.org/doc/manuals/debian-reference/ch02.en.html#_debian_package_file_names

![Screen Shot 2021-04-06 at 9 12 16 AM](https://user-images.githubusercontent.com/25510/113743149-3149c880-96b8-11eb-8187-5663f4b98ca3.png)

Reply:
If we strip everything that isn't allowed, that should protect against weird unicode char insertion as well 👍

Reply:
I would definitely go with this approach (allowed list of characters in packages). That should stop a bunch of shenanigans from unicode to paths to installing from a URL.

Reply:
> Isn't this still possible by doing repository/my/path/package.deb as an item?

Nope, it will try to locate that package in the ubuntu repos.

I'll update this to validate with allowed chars.